conix.nix

```diff
@@ -0,0 +1,150 @@
+self: super:
+{ conix = (super.conix or {}) // rec
+{
+  emptyModule = {};
+
+  mergeModule = 
+    # AttrSet -> AttrSet -> AttrSet
+    self.lib.attrsets.recursiveUpdate;
+
+  mergeModuleFunc 
+    # (AttrSet -> AttrSet) -> (AttrSet -> AttrSet) -> AttrSet -> AttrSet
+    = mkModuleA: mkModuleB: x: 
+    mergeModule (mkModuleA x) (mkModuleB x);
+
+  # FoldMap for module functions 
+  foldMap 
+    # (a -> AttrSet -> AttrSet) -> [a] -> AttrSet -> AttrSet
+    = f: 
+      builtins.foldl' (m: x: mergeModuleFunc m (f x)) (x: {});
+
+  # Fold for module functions
+  fold
```

Comment:
Perhaps call functions of type `AttrSet -> AttrSet` a "page" so that this name can be more specific (i.e. `foldPages`)